Detectors/ITSMFT/ITS/workflow/src/ThresholdAggregatorSpec.cxx

```diff
@@ -0,0 +1,220 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   ThresholdAggregatorSpec.cxx
+
+#include "ITSWorkflow/ThresholdAggregatorSpec.h"
+
+#ifdef WITH_OPENMP
+#include <omp.h>
+#endif
+
+namespace o2
+{
+namespace its
+{
+
+//////////////////////////////////////////////////////////////////////////////
+// Default constructor
+ITSThresholdAggregator::ITSThresholdAggregator()
+{
+  mSelfName = o2::utils::Str::concat_string(ChipMappingITS::getName(), "ITSThresholdAggregator");
+}
+
+//////////////////////////////////////////////////////////////////////////////
+void ITSThresholdAggregator::init(InitContext& ic)
+{
+  LOGF(info, "ITSThresholdAggregator init...", mSelfName);
+
+  this->mVerboseOutput = ic.options().get<bool>("verbose");
+  this->mCcdbUrl = ic.options().get<std::string>("ccdb-url");
+
+  return;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+// Main running function
+// Get DCSconfigObject_t from EPNs and aggregate them in 1 object
+void ITSThresholdAggregator::run(ProcessingContext& pc)
+{
+
+  //take run type and scan type at the beginning
+  if (this->mRunType == -1) {
+    this->mRunType = pc.inputs().get<short int>("runtype");
+    this->mScanType = pc.inputs().get<char>("scantype");
+    this->mFitType = pc.inputs().get<char>("fittype");
+    this->updateRunID(pc); // run number
+  }
+
+  if (this->mLHCPeriod.empty()) {
+    this->updateLHCPeriod(pc);
+  }
+
+  // Read strings with tuning info
+  const auto tunString = pc.inputs().get<o2::dcs::DCSconfigObject_t>("tunestring");
+  std::string tmpString(tunString.begin(), tunString.end());
+  //Merge all strings coming from several sources (EPN)
+  std::copy(tmpString.begin(), tmpString.end(), std::back_inserter(tuningMerge));
+
+  return;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+void ITSThresholdAggregator::finalize(EndOfStreamContext* ec)
+{
+  LOGF(info, "endOfStream report:", mSelfName);
+
+  // Below is CCDB stuff
+  long tstart, tend;
+  tstart = o2::ccdb::getCurrentTimestamp();
+  constexpr long SECONDSPERYEAR = 365 * 24 * 60 * 60;
+  tend = o2::ccdb::getFutureTimestamp(SECONDSPERYEAR);
```

Comment:
note that the `tstart = o2::ccdb::getCurrentTimestamp()` gives time in `ms`, while your `tend` is just 1 year in seconds, while it should be an abs. time-stamp in ms. If you want to assign 1y validity, you should have
```
tend = tstart + 365L*24*3600*1000;
```